modules/common/job/job.go

```diff
@@ -258,14 +258,20 @@ func (j *Job) waitOnJob(
 		return ctrl.Result{}, nil
 	} else if j.actualJob.Status.Failed > 0 {
 		if existingJobHash != j.hash {
-			h.GetLogger().Info(
-				"The hash of the job changed but the previous failed job still exists. " +
-					"Deleting old job and requeueing.")
-			err := DeleteJob(ctx, h, j.actualJob.Name, j.actualJob.Namespace)
-			if err != nil {
-				return ctrl.Result{}, err
+			if j.preserve {
+				h.GetLogger().Info(
+					"The hash of the job changed and the previous failed job still exists. " +
+						"Consider deleting the old job and requeue.")
```

Comment:
This is problematic. This means if a job failed  then the operator requesting a different config for the job will not get applied but simply ignored.

Reply:
I agree, what I really wanted was a way to signal to the job owner that the job was deleted

Reply:
OTOH, deleting a job means deleting pod logs, I wonder if could keep the failed logs somehow